jdk-profiles/pom.xml

```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Copyright (c) 2023, NVIDIA CORPORATION.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>com.nvidia</groupId>
+        <artifactId>rapids-4-spark-parent_2.12</artifactId>
+        <version>23.12.0-SNAPSHOT</version>
+    </parent>
+    <groupId>com.nvidia</groupId>
+    <artifactId>rapids-4-spark-jdk-profiles_2.12</artifactId>
```

Comment:
The premerge CI definitely depends on the activeByDefault profiles because they run integration tests for the default lowest supported version. When we drop support for these versions, if the activeByDefault goes away, we would need to specify the version each time. 

I would suggest if we remove activeByDefault, we should add a corresponding profile for the default version, or maybe simply use the first noSnapshot version, and update the premerge scripts to specify that buildver whenever running the integration tests.

I'm not sure what else in CI does depend on activeByDefault, but we should figure that out as well.

Reply:
Per offline conversations this is a compromise/workaround focusing on eliminating the need to specify the jdk profiles explicitly. 

This PR demonstrates how it can be solved preserving the default buildver profiles.

A nice feature of the existing setup is that the external developers coming from outside can execute the first build with mvn package without reading the manual.

If we decide to ditch this feature we can put in enforcer rules educating the developers about our quirks.